packages/completer/src/tokens.ts

```diff
@@ -81,12 +103,71 @@ export const ICompletionProviderManager = new Token<ICompletionProviderManager>(
 );
 
 export interface ICompletionProviderManager {
+  /**
+   * Register a completer provider with the manager.
+   *
+   * @param {ICompletionProvider} provider - the provider to be registered.
+   */
   registerProvider(provider: ICompletionProvider): void;
-  setTimeout(timeout: number): void;
+
+  /**
+   * Invoke the completer in the widget with provided id.
+   *
+   * @param {string} id - the id of notebook panel, console panel or code editor.
+   */
+  invoke(id: string): void;
+
+  /**
+   * Activate `select` command in the widget with provided id.
+   *
+   * @param {string} id - the id of notebook panel, console panel or code editor.
+   */
+  select(id: string): void;
+
+  /**
+   * Attach completion service to a widget.
+   *
+   * @param widget - the widget which requests completion service.
+   * @param completerContext - current completion context.
+   */
+  attachWidget(
```

Comment:
I would change the wording to

```suggestion
  attachCompleter(
```